inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+            } catch (SolrServerException e) {
+                e.printStackTrace();
+            }
+
+            return result;
+        }
+    }
+
+    /**
+     * Search a Solrdocument by id and return the text
+     * @param aRepository Solr repository
+     * @param aTraits Request parameters
+     * @param aCollectionId Name of the collection / index
+     * @param aDocumentId unique id of the document
+     * @return the text of the document
+     * @throws SolrException wrong id or connection issues
+     */
+    @Override
+    public String getDocumentText(DocumentRepository aRepository,
+                                  SolrSearchProviderTraits aTraits,
+                                  String aCollectionId, String aDocumentId)
+            throws IOException
+    {
+        if (!aCollectionId.equals(aTraits.getIndexName()))
+        {
+            throw new IllegalArgumentException(
+                    "Requested collection name does not match connection collection name");
+        }
+
+        SolrQuery getQuery = new SolrQuery(DOC_ID_KEY + ":" + aDocumentId);
+        getQuery.setRows(1);
+
+        HttpSolrClient client = makeClient(aTraits);
+
+        QueryResponse response = new QueryResponse();
+        SolrDocumentList documents = new SolrDocumentList();
+        SolrDocument document = new SolrDocument();
+
+        getQuery.setRows(1);
+        getQuery.setParam("collection", aTraits.getIndexName());
+        getQuery.setParam("qt", aTraits.getSearchPath());
+
+        try {
+            response = client.query(getQuery);
+            documents = response.getResults();
+            document = documents.get(0);
+        } catch (SolrException | SolrServerException e) {
+            e.printStackTrace();
+        }
+        return (String) document.getFirstValue(aTraits.getTextField());
+
+    }
+
+
+    @Override
+    public InputStream getDocumentAsStream(DocumentRepository aRepository,
+                                           SolrSearchProviderTraits aTraits,
+                                           String aCollectionId, String aDocumentId)
+            throws IOException {
+        return IOUtils.toInputStream(
+                getDocumentText(aRepository, aTraits, aCollectionId, aDocumentId), UTF_8);
+    }
+
+    @Override
+    public String getDocumentFormat(DocumentRepository aRepository,
+                                    SolrSearchProviderTraits aTraits,
+                                    String aCollectionId, String aDocumentId)
+    {
+        return TextFormatSupport.ID;
+    }
+
+    /**
+     * Create à client and connect it to the server
+     * @param aTraits parameters of the connection
+     * @return HttpSolrClient object use synchrone methode
+     */
+    private HttpSolrClient makeClient(SolrSearchProviderTraits aTraits)
+            throws HttpHostConnectException
+    {
+
+        return new HttpSolrClient.Builder(aTraits.getRemoteUrl() + "/solr")
```

Comment:
Why not have `solr` in the user-provided URL? If the user wants to connect to a SOLR server which uses a non-default URL layout, this might make it impossible.